src/core/server/worker_threads/worker_threads_service.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { CoreService } from 'src/core/types';
+import path from 'path';
+import Piscina from 'piscina';
+import { InternalElasticsearchServiceSetup, IScopedClusterClient } from '../elasticsearch';
+
+export interface SetupDeps {
+  elasticsearch: InternalElasticsearchServiceSetup;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface WorkerThreadsServiceSetup {}
+
+export interface RunOptions<T> {
+  name: string;
+  args: T;
+  scopedClusterClient?: IScopedClusterClient;
+  abortSignal?: AbortSignal;
+}
+
+export interface WorkerThreadsServiceStart {
+  run: <T = unknown, U = unknown>({
+    name,
+    args,
+    abortSignal,
+    scopedClusterClient,
+  }: RunOptions<T>) => Promise<U>;
+}
+
+export class WorkerThreadsService
+  implements CoreService<WorkerThreadsServiceSetup, WorkerThreadsServiceStart>
+{
+  private piscina?: Piscina;
+
+  public async setup({ elasticsearch }: SetupDeps) {
+    this.piscina = new Piscina({
+      filename: path.resolve(__dirname, 'worker.js'),
+      workerData: elasticsearch.getCreateClusterClientParams('worker_thtread'),
+    });
+    return {};
+  }
+  public async start() {
+    return {
+      run: async <T, U>({ name, args, abortSignal, scopedClusterClient }: RunOptions<T>) =>
+        this.piscina!.run(
+          { args, scopedHeaders: scopedClusterClient?.headers },
```

Comment:
You can run a task with a scoped Elasticsearch client, when you do that we copy the headers from the passed in scopedClient and use that on the worker to construct a scoped cluster client for the specific task being run.

Probably this is a bit naive and there's some edge cases that need to be thought through.